sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractProjectConfigurationTest.java

```diff
@@ -44,110 +43,87 @@
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(config.roslynReportPaths()).containsOnly(workDir.resolve("roslyn-report.json"), workDir.resolve("roslyn-report2.json"));
   }
 
   @Test
   public void onlyOldRoslynReportPresent() throws IOException {
-    createOldRoslynOut();
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Path path = temp.newFile("roslyn-report.json").toPath();
+
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.get("sonar.cs.roslyn.reportFilePath")).thenReturn(Optional.of(path.toString()));
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(config.roslynReportPaths()).containsOnly(workDir.resolve("roslyn-report.json"));
   }
 
   @Test
   public void giveWarningsWhenGettingProtobufPathAndNoPropertyAvailable() {
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(logTester.logs(LoggerLevel.WARN)).containsOnly("Property missing: 'sonar.cs.analyzer.projectOutPaths'. No protobuf files will be loaded for this project.");
   }
 
   @Test
   public void noWarningsWhenGettingProtobufPathAndNoPropertyAvailable_TestProject() {
     // Test projects have sonar.tests property set, main projects don't
-    settings.setProperty("sonar.tests", "");
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.hasKey("sonar.tests")).thenReturn(true);
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).isEmpty();
     assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
   }
 
   @Test
   public void giveWarningsWhenGettingProtobufPathAndNoFolderAvailable() throws IOException {
     Path path1 = createProtobufOut("report");
-    settings.setProperty("sonar.cs.analyzer.projectOutPaths", new String[] {path1.toString(), "non-existing"});
-    config = new AbstractProjectConfiguration(settings.asConfig(), "cs") {
-    };
+
+    Configuration configuration = createEmptyConfiguration();
+    when(configuration.getStringArray("sonar.cs.analyzer.projectOutPaths")).thenReturn(new String[] {path1.toString(), "non-existing"});
+
+    AbstractProjectConfiguration config = createAbstractProjectConfiguration(configuration);
     assertThat(config.protobufReportPaths()).containsOnly(path1.resolve("output-cs"));
     assertThat(logTester.logs(LoggerLevel.WARN)).hasSize(1);
```

Comment:
I know it's not related to your change, but clean as you code 
I think you can replace `hasSize(1)` and `get(0).matches` with `hasOnlyOneElementSatisfying`

Reply:
same for lines 117-118; 143-144;